modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -0,0 +1,205 @@
+        break;
+      case 'MultiPoint':
+        pointPositions += geometry.coordinates.length;
+        geometry.coordinates.forEach(point => {
+          if (point.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'LineString':
+        linePositions += geometry.coordinates.length;
+        linePaths++;
+
+        geometry.coordinates.forEach(coord => {
+          if (coord.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'MultiLineString':
+        geometry.coordinates.forEach(line => {
+          linePositions += line.length;
+          linePaths++;
+
+          line.coordinates.forEach(coord => {
+            if (coord.length === 3) maxCoordLength = 3;
+          });
+        });
+        break;
+      case 'Polygon':
+        polygonObjects++;
+        polygonRings += geometry.coordinates.length;
+        polygonPositions += geometry.coordinates.flat(1).length;
+
+        geometry.coordinates.flat().forEach(coord => {
+          if (coord.length === 3) maxCoordLength = 3;
+        });
+        break;
+      case 'MultiPolygon':
+        geometry.coordinates.forEach(polygon => {
+          polygonObjects++;
+          polygonRings += polygon.length;
+          polygonPositions += polygon.flat(1).length;
+
+          geometry.coordinates.flat().forEach(coord => {
+            if (coord.length === 3) maxCoordLength = 3;
+          });
+        });
+        break;
+      default:
+        throw new Error(`Unsupported geometry type: ${geometry.type}`);
+    }
+  });
+
+  return {
+    pointPositions,
+    linePositions,
+    linePaths,
+    coordLength: maxCoordLength,
+    polygonPositions,
+    polygonObjects,
+    polygonRings
+  };
+}
+
+function secondPass(features, firstPassData, options = {}) {
+  const {
+    pointPositions,
+    linePositions,
+    linePaths,
+    coordLength,
+    polygonPositions,
+    polygonObjects,
+    polygonRings
+  } = firstPassData;
+  const {PositionDataType = Float32Array} = options;
+  const points = {
+    positions: new PositionDataType(pointPositions * coordLength),
+    objectIds: new Uint32Array(pointPositions)
+  };
+  const lines = {
+    pathIndices: new Uint32Array(linePaths),
+    positions: new PositionDataType(linePositions * coordLength),
+    objectIds: new Uint32Array(linePositions)
+  };
+  const polygons = {
+    polygonIndices: new Uint32Array(polygonObjects),
+    primitivePolygonIndices: new Uint32Array(polygonRings),
+    positions: new PositionDataType(polygonPositions * coordLength),
+    objectIds: new Uint32Array(polygonPositions)
+  };
+
+  const index = {
+    pointPosition: 0,
+    linePosition: 0,
+    linePath: 0,
+    polygonPosition: 0,
+    polygonObject: 0,
+    polygonRing: 0,
+    feature: 0
+  };
+
+  features.forEach(feature => {
+    const geometry = feature.geometry;
+
+    switch (geometry.type) {
+      case 'Point':
+        handlePoint({coords: geometry.coordinates, points, index, coordLength});
```

Comment:
Using named parameters here generates a temporary small object for each feature, putting pressure on the garbage collector. 

Since this is not a user facing function, the value of named parameters is less, so we typically avoid them and just use positional parameters in these cases.